### PR TITLE
preliminary ECDH support

### PR DESCRIPTION
**Pull request by maxtaco**
- make sure that we can import ECDH private keys
- make some test cases
- write out some test data
- start on actual ECDH decryption, but the crux is unimplemented

### CODE REVIEW

**Comment by maxtaco** [comment]
@oconnor663 can you take a look? 

This allows us to import and pass through ECDH keys, but I didn't implement decryption with ECDH keys.

**Comment by maxtaco** [comment]
With this fix, we're now able to run `keybase pgp select` on keys that have ECDH subkeys (over Nist or Brainpool curves). Whereas before we weren't

File: openpgp/packet/private_key.go [hunk 1]
```diff
@@ -366,6 +374,24 @@ func (pk *PrivateKey) parseElGamalPrivateKey(data []byte) (err error) {
 	return nil
 }
 
+func (pk *PrivateKey) parseECDHPrivateKey(data []byte) (err error) {
```
Comment (oconnor663) [inline openpgp/packet/private_key.go:377]: This function looks like it's a duplicate of the ones around it. It looks like the only difference between them is the types they cast into? Is that a "don't fix what ain't broke" thing?

**Comment by oconnor663** [comment]
Looks fine to me, with the one question above.

File: openpgp/packet/private_key.go (hunk 1 above)
Comment (maxtaco) [reply openpgp/packet/private_key.go:377]: Yeah, it at least will allow reading of the key without crashing (what we used to do with an actual `panic`). I ought to go ahead and implement ECDH but it's tricky and takes a lot of tinkering, so I think it's worth it to put off for now, and at least get import/export of these keys working.

**Pull request merged**